belief_manager.py

```diff
@@ -0,0 +1,578 @@
+              # Use this one.
+              del associations[best_transmitter]
+            else:
+              # Otherwise, use the other one.
+              continue
+
+          associations[transmitter] = reading
+          self._cycle_data[transmitter] = self._cycles
+          associated = True
+          best_center_distance = center_distance
+          best_transmitter = transmitter
+
+      if (not associated and not weak):
+        # It fit inside none of our previous regions.
+        logger.info("Asuming %s is new transmitter." % (str(reading)))
+        new.append(reading)
+        new_index = 4 + self._filter.number_of_transmitters() + len(new)
+        self._cycle_data[new_index] = self._cycles
+
+    logger.debug("Associated bearings with transmitters: %s" % (associations))
+    return (associations, new)
+
+  def _distance_from_strength(self, readings):
+    """ Takes a set of transmitter readings along with their strengths and uses
+    their strengths to guess how far away they are. This is not a very accurate
+    method of guaging transmitter positions, but it is used initially, because
+    it works with only one reading.
+    Args:
+      readings: A set of associated LOB readings, or a list of LOB, strength
+      tuples.
+    Returns: Either a dictionary where each key is a transmitter index, and each
+      value is a tuple of the X and Y position of the transmitter, or, if
+      non-associated readings were provided, a simple list of these tuples. """
+    if type(readings) == dict:
+      positions = {}
+      for transmitter, reading in readings.iteritems():
+        positions[transmitter] = \
+            self._single_point_strength_distance(self._filter.position(),
+                                                 reading[0], reading[1])
+    else:
+      positions = []
+      for reading in readings:
+        location = self._single_point_strength_distance(self._filter.position(),
+                                                        reading[0], reading[1])
+        positions.append(location)
+
+    return positions
+
+  def _distance_from_past_states(self, transmitters):
+    """ Uses data from previous states as a more accurate method for calculating
+    transmitter positions. The way this works is that the drone gets two LOBs on
+    the transmitter at two different times. Where those LOBs cross is the
+    location of the transmitter.
+    transmitters: The indexes of the transmitters in the state to use.
+    Returns: A dict of tuples, with each tuple containing the estimated x and y
+      position of the transmitter. The keys are the indices of the corresponding
+      transmitters in the state. """
+    logger.debug("Past lobs: Calculating distance.")
+    positions = {}
+    for transmitter in transmitters:
+      # Check whether we have enough data to use the filtered state.
+      current_lob = self._filter.state()[transmitter]
+      current_position = self._filter.position()
```

Comment:
Once again, it's not very clear what exactly you're referring to. In this case, I'm going to assume it's the bottom line, in which case, it does change every iteration. All the position() method does is it goes and pulls out the position from the filtered state in the Kalman filter. The comment above it, however, is outdated and confusing, and I have taken the liberty of fixing it.